app/2.0/docs/devguide/cross-scope-styling.md

```diff
@@ -0,0 +1,331 @@
+
+These custom properties can be defined similarly to other standard CSS properties
+and will inherit from the point of definition down the composed DOM tree,
+similar to the effect of `color` and `font-family`.
+
+In the simple example below, the author of `my-toolbar` identified the need for
+users of the toolbar to be able to change the color of the toolbar title.  The
+author exposed a custom property called `--my-toolbar-title-color` which is
+assigned to the `color` property of the selector for the title element.  Users
+of the toolbar may define this variable in a CSS rule anywhere up the tree, and
+the value of the property will inherit down to the toolbar where it is used if
+defined, similar to other standard inheriting CSS properties.
+
+Example: { .caption }
+
+```html
+<dom-module id="my-toolbar">
+  <template>
+    <style>
+      :host {
+        padding: 4px;
+        background-color: gray;
+      }
+      .title {
+        color: var(--my-toolbar-title-color);
+      }
+    </style>
+    <span class="title">{{title}}</span>
+  </template>
+  <script>
+    class MyToolbar extends Polymer.Element {
+      static get is() {
+        return "my-toolbar";
+      }
+    }
+    customElements.define(MyToolbar.is, MyToolbar);
+</script>
+</dom-module>
+```
+
+Example usage of `my-toolbar`: { .caption }
+
+```html
+<dom-module id="my-element">
+  <template>
+    <style>
+      /* Make all toolbar titles in this host green by default */
+      :host {
+        --my-toolbar-title-color: green;
+      }
+      /* Make only toolbars with the .warning class red */
+      .warning {
+        --my-toolbar-title-color: red;
+      }
+    </style>
+    <my-toolbar title="This one is green."></my-toolbar>
+    <my-toolbar title="This one is green too."></my-toolbar>
+    <my-toolbar class="warning" title="This one is red."></my-toolbar>
+  </template>
+  <script>
+    class MyElement extends Polymer.Element {
+      static get is() {
+        return "my-element";
+      }
+    }
+    customElements.define(MyElement.is, MyElement);
+  </script>
+</dom-module>
+```
+
+The `--my-toolbar-title-color` property only affects the color of the title
+element encapsulated in `my-toolbar`'s internal implementation.  In the
+future the `my-toolbar` author can rename the `title` class or
+restructure the internal details of `my-toolbar` without changing the custom
+property exposed to users.
+
+You can also include a default value in the `var()` function, to use in case the user
+doesn't set the custom property:
+
+```css
+color: var(--my-toolbar-title-color, blue);
+```
+
+Thus, custom CSS properties are a powerful way for element authors to
+expose a theming API to their users in a way that naturally fits right alongside
+normal CSS styling. 
+
+### Custom CSS mixins
+
+It may be tedious (or impossible) for an element author to predict every
+CSS property that may be important for theming, let alone expose every
+property individually.
```

Comment:
CSS mixins are a proposal that has not been implemented by any of the browsers. 
To use them, you need to import an optional polyfill. 

Any element using mixins should import the polyfill. 

```
<!-- import CSS mixins polyfill -->
<link rel="import" href="/bower_components/shadycss/apply-shim.html">
```
For backward compatibility, the `polymer.html` import includes the CSS mixins polyfill. No extra import is required when defining hybrid elements.